libcst/tests/pyre/simple_class.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# fmt: off
+from typing import Sequence
+
+
+class Item:
+    def __init__(self, n: int) -> None:
+        self.number = n
```

Comment:
We expect pyre to infer `self.number` has type int but it's not.

Reply:
The reason pyre doesn't infer `int` for `self.number` here is because this attribute was not given an annotation (this would be an error in strict: "Missing attribute annotation [4]: Attribute `number` of class `Item` has type `int` but no type is specified.") 

It's true that at the moment of assignment we are sure that `self.number` is an integer. But would it be more confusing to special case this, when any later reference to `self.number` will be unknown? Maybe consistency is better here.

Note: Pyre does infer that if you assign an attribute like:
- a literal, ie.`self.x = 1`
- an annotated parameter with the same name, ie. `self.number = number`

Then Pyre will assume that the annotation will always be of the type at assignment unless otherwise specified. In these cases, attempting to assign the attribute to something of a different type later on will yield a type error, the same way as it would for an attribute declared like `self.number: int = n`. 
This is where we draw the line, because when the value field gets more complex some or parts of the type could be unknown, and it becomes less safe or consistent for pyre to assume an explicit annotation in cases where it happens to know the type.

Let me know if this makes sense! Will move on to addressing the other issues brought up for now, but happy to revisit this one. :)

Reply:
This test file didn't enable pyre strict mode. So, Pyre doesn't complain self.number is missing type. The explanation of missing type annotation make sense. We can probably assume all code will be in pyre strict mode and I can update the test case as pyre-strict.